spec/lib/barcelona/chaos_spec.rb

```diff
@@ -0,0 +1,46 @@
+require "rails_helper"
+
+describe Barcelona::Chaos do
+  describe "#run" do
+    let(:district) { create :district }
+    let(:asg_mock) { double }
+    let(:chaos) { described_class.new(district, 1) }
+
+    before do
+      allow_any_instance_of(AwsAccessor).to receive(:autoscaling) { asg_mock }
+    end
+
+    context "when the district has only active instances" do
+      before do
+        expect(district).to receive(:container_instances) do
+          [
+            {status: 'ACTIVE', ec2_instance_id: 'i-11111111'}
+          ]
+        end
+      end
+
+      it "makes a terminate request for the instance" do
+        expect(asg_mock).to receive(:terminate_instance_in_auto_scaling_group).with(
+                              instance_id: 'i-11111111',
+                              should_decrement_desired_capacity: false
+                            )
+        chaos.run
+      end
+    end
```

Comment:
It might be worthwhile to test the case where there are multiple instances, and you only terminate a subset of them. A small code change could pass this spec but actually break that functionality.